guidelines/design-guidelines.md

```diff
@@ -31,4 +45,156 @@ Example:
 var transportConfig = endpointConfig.UseTransport<MsmqTransport>();
 
 transportConfig.MsmqLabelGenerator(context => return $"{context.Headers['NServiceBus.EnclosedMessageTypes']}");
-```
\ No newline at end of file
+```
+
+### FAQ
+
+##### How do I apply changes without recompiling?
+
+We will be providing additional packages outside the core which allows to hook in configuration sections or alike to support previous scenarios if necessary. Another approach is to show in samples how to load configuration by using `AppSettings`, `ConfigurationManager`, etc.
+
+## Startable and stoppable components
+
+```
+class Component
+{
+  Task Start();
+  Task Stop();
+}
+```
+
+* All components externally implemented but managed by the core (pipelines, satellites...) should try to gracefully stop, if they can't they should hang
+* The core will *not* timebox any start or stop operations to allow proper debugging and analysis of misbehaving components
+* Stop operations should be done concurrently if possible/sensible to allow all components to initiate a graceful stop
+* `CancellationToken` and `CancellationTokenSource` are an implementation detail of the component that needs to stop and are not managed by the core
+
+Translating these principles into code, here is how the core handles startable and stoppable components:
+
+```
+var components = new List<Component>();
+// Start
+foreach(cmp in components)
+{
+  await cmp.Start().ConfigureAwait(false);
+}
+
+// Stop
+var stopTasks = components.Select(cmp => cmp.Stop());
+await Task.WhenAll(stopTasks);
+```
+
+## Composition
+
+Goal: The public API of NServiceBus is a composition API consisting out of multiple capabilities. The APIs in the core are extensible enough so that [capabilities](https://github.com/Particular/Vision/labels/Capability) can extend the composition API without needing to share the same release cycle as the core, reducing the coupling and allows us to organize code in a more cohesive way.
```

Comment:
> Goal: ~The~ *the* public API of NServiceBus is a composition API consisting ~out~ of multiple capabilities. The APIs in the *NServiceBus* core *package* are extensible enough so that [capabilities](https://github.com/Particular/Vision/labels/Capability) can extend the composition API without needing to share the same release cycle as the core, reducing the coupling and ~allows~ *allowing* us to organize code in a more cohesive way.